neo3/wallet/account.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from neo3 import contracts, settings, wallet
+from neo3.core import cryptography, types
+
+
+class Account:
+
+    def __init__(self, script_hash: types.UInt160, nep2key: str = None):
+        self._script_hash: types.UInt160 = script_hash
+        self._nep2key: str = nep2key
+        self.label = ''
+        self.is_default = False
+        self.lock = False
+        self._key: cryptography.KeyPair = None
+        self.contract: contracts.Contract = None
+        self.extra = None
+
+    @property
+    def address(self) -> str:
+        return wallet.to_address(self._script_hash, settings.network.account_version)
+
+    @property
+    def decrypted(self) -> bool:
+        return self._nep2key is not None or self._key is not None
+
+    @property
+    def has_key(self) -> bool:
```

Comment:
Without looking at the code the name has me wondering what key we're talking about. Public key, private key, keypair? With looking at the code I wonder why the check is not the same as the current `decrypted` property.

Reply:
The name `has_key()` was taken from [Neo's code](https://github.com/neo-project/neo/blob/604541c91b3151ee40a8613cb8534ebc0540fd27/src/neo/Wallets/NEP6/NEP6Account.cs#L16). I realize now that there was a typo at `decrypted`, it should have been `return self._nep2key is None or self._key is not None`.